evaluator/builtin_string.go

```diff
@@ -50,6 +50,60 @@ func builtinLength(args []types.Datum, _ context.Context) (d types.Datum, err er
 	}
 }
 
+// See https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_bin.
+func builtinBIN(args []types.Datum, ctx context.Context) (d types.Datum, err error) {
+	var b []byte
+	var bs string
+	switch args[0].Kind() {
+	case types.KindNull:
+		return d, nil
+	default:
+		v, err := args[0].ToInt64(ctx.GetSessionVars().StmtCtx)
+		if err != nil {
+			return d, errors.Trace(err)
+		}
+		switch {
```

Comment:
You don't need to convert an integer to bytes, just use bit operator :)
> 
while(v > 0){
    if (v & 1 > 0){
         bs = bs + '1'
    } else {
        bs = bs + '0'
    }
    v = v >> 1
}